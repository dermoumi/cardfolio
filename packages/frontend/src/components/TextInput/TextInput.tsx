import type { FC } from "react";

export type TextInputProps = {
  value: string;
  type: "text" | "password" | "search";
  onChange: (value: string) => void;
  placeholder?: string;
};

const TextInput: FC<TextInputProps> = ({ value, type, onChange, placeholder }) => {
  return (
    <input
      type={type}
      value={value}
      onChange={(e) => onChange(e.target.value)}
      placeholder={placeholder}
    />
  );
};

export default TextInput;
