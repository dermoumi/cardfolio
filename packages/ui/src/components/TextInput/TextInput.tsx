import type { FC } from "react";

import styles from "./TextInput.module.css";

export type TextInputProps = {
  name: string;
  value?: string;
  type?: "text" | "password" | "search";
  onChange?: (value: string) => void;
  placeholder?: string;
};

const TextInput: FC<TextInputProps> = ({ value, name, type = "text", onChange, placeholder }) => {
  return (
    <input
      className={styles.textInput}
      name={name}
      type={type}
      value={value}
      onChange={(e) => onChange?.(e.target.value)}
      placeholder={placeholder}
    />
  );
};

export default TextInput;
