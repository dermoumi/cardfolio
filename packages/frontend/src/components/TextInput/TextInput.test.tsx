import { fireEvent, render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import TextInput from ".";

describe("TextInput", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<TextInput placeholder="Enter text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onChange when text is entered", () => {
    const handleChange = vi.fn();
    const { getByPlaceholderText } = render(
      <TextInput placeholder="Enter text" onChange={handleChange} />,
    );

    const input = getByPlaceholderText("Enter text") as HTMLInputElement;
    fireEvent.change(input, { target: { value: "Hello" } });

    expect(handleChange).toHaveBeenCalled();
    expect(input.value).toBe("Hello");
  });
});
