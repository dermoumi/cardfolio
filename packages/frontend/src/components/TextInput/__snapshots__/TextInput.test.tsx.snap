// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextInput > matches snapshot 1`] = `
<DocumentFragment>
  <input
    placeholder="Enter text"
    type="text"
  />
</DocumentFragment>
`;
